src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -505,13 +488,29 @@ internal sealed class AnalyzerInconsistencyBuildResponse : BuildResponse
 
     internal sealed class RejectedBuildResponse : BuildResponse
     {
+        public string Reason;
+
         public override ResponseType Type => ResponseType.Rejected;
 
+        public RejectedBuildResponse(string reason)
```

Comment:
This is the most impactful part of the change because it forces us to declare a reason every time we reject a build. Note that rejecting a build is different than a build failing. A build failing is a completed build (it just failed). A rejected build is one the server said "i can't handle this, fall back to csc / vbc".